lib/app/features/home/home_tabs/score_tab/cubit/score_tab_cubit.dart

```diff
@@ -14,17 +14,22 @@ class ScoreTabCubit extends Cubit<ScoreTabState> {
   final HomeScreenRepository homeScreenRepository;
 
   Future<void> fetchMatchesByDate(DateTime date) async {
-    String formattedDate = DateFormat('yyyy-MM-dd').format(date);
-    emit(const LoadingMatchesState());
-    if (date.isBefore(DateTime.now()) || date.isAtSameMomentAs(DateTime.now())) {
-      try {
-        final matchesByDate = await homeScreenRepository.fetchMatchesByDate(AppConstVariables.fullTime, formattedDate);
-        if (matchesByDate!.isNotEmpty) {
+    String formattedStringDate = DateFormat('yyyy-MM-dd').format(date);
```

Comment:
Do you need these parsed dates elsewhere? If you only check this condition below, you can do it much more simple.
Get rid of:
`String formattedStringDate`
`DateTime formattedDate`
`DateTime formattedActualDate`

Instead:
`final dateNow = DateTime.now()`
`if (date.isBefore(dateNow) || date.isAtSameMomentAs(dateNow) {...}`

Reply:
`formattedStringDate` is needed to parse it as query for Api, because it requires 'yyyy-MM-dd' form;
`DateTime formattedDate` and `DateTime formattedActualDate` were made like that in order to compare only dates without hour, your solution returns date and actual hour so it may cause unwanted differences in display

Reply:
Ok, but I think this 
`DateTime formattedActualDate = DateFormat('yyyy-MM-dd').parse(DateFormat('yyyy-MM-dd').format(DateTime.now()));` 
is not necessary. Can change it to:
`DateTime formattedActualDate = DateFormat('yyyy-MM-dd').parse(DateTime.now()); `

Reply:
then `formattedActualDate` needs to be a String type so it makes complications with comparing two values later

Reply:
Ok, I get it now. But maybe extract this to an extension function. You do this thing twice in different methods so it makes sense and here it's quite unreadable. Especially `formattedActualDate`.